runtime/vm/resolvesupport.cpp

```diff
@@ -601,6 +601,8 @@ resolveStaticFieldRefInto(J9VMThread *vmStruct, J9Method *method, J9ConstantPool
 		UDATA modifiers = 0;
 		J9Class *localClassAndFlags = NULL;
 		UDATA initStatus = 0;
+		J9Class *currentClassFromCP = NULL;
```

Comment:
can you define this in the scope that it's used in

Reply:
The compiler complains because of the 'illegalAccess' `goto` which is shortly after it

```
resolvesupport.cpp: In function 'void* resolveStaticFieldRefInto(J9VMThread*, J9Method*, J9ConstantPool*, UDATA, UDATA, J9ROMFieldShape**, J9RAMStaticFieldRef*)':
resolvesupport.cpp:675: error: jump to label 'illegalAccess'
resolvesupport.cpp:618: error:   from here
resolvesupport.cpp:669: error:   crosses initialization of 'J9Class* currentDefiningClass'
resolvesupport.cpp:668: error:   crosses initialization of 'J9Class* currentClassFromCP'
```

Should I take out the variable definitions and rewrite the check as:
```
if ((J9_CURRENT_CLASS(classFromCP) !=  J9_CURRENT_CLASS(definingClass)) && (J9ROMCLASS_IS_UNSAFE(v->romClass) == 0)) { ... }
```

Reply:
That's odd.  Can you work around this by introducing a scope inside the if statement that defines and uses the locals?

Reply:
I restructured my changes so that the 'is this the same class' check occurs once within `finalFieldSetAllowed(...)` instead of within `resolveInstanceFieldRefInto(...)` and `resolveStaticFieldRefInto(...)`. 

The way that the VM handles illegal access to final fields is slightly different than the way it handles other illegal access (eg. non module access). Since the VM checks that a final field is accessed within a constructor occurs in `finalFieldSetAllowed`, I think it makes more sense to check at that stage if it is within the right constructor. 

I pushed the new changes, but am still in the process of retesting them